z3c/dependencychecker/db.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+import logging
+
+
+logger = logging.getLogger(__name__)
+
+
+class ImportsDatabase(object):
+
+    def __init__(self):
+        self._requirements = []
+        self._extras_requirements = {}
+
+    def set_requirements(self, requirements_list):
+        for requirement in requirements_list:
+            if self._requirement_already_exists(requirement):
+                logger.warn(
+                    '"%s" is declared twice on setup.py',
+                    requirement.name,
+                )
+            else:
+                self._requirements.append(requirement)
+
+    def _requirement_already_exists(self, requirement):
+        names = self.get_requirements_names()
+        return bool(requirement.name in names)
+
+    def get_requirements_names(self):
```

Comment:
This is one of the points where our coding tastes differ. And thus one of the points where I'm getting a bit philosophical: "do I really dislike this? could I use this? is this handy?"

I've always considered `get_xyz()` and `set_xyz()` methods to be a java naming convention. Instead of `.get_xyz()`, the python style is to do just `.xyz()`. And setting something is "just" assigning something to it.

So.... this isn't a request to change your code, it is more of an observation. I'm currently evaluating whether it might actually *help* sometimes to name getters and setters explicitly.

Reply:
The important part here is that we are returning a list of plain strings rather than ``DottedName`` objects, which is what self._requirements is storing. Now that you mention it, yes ``requirement_names`` sounds better, I will change it.